### PR TITLE
feat: validate uploaded image files

### LINKED ISSUE #3755: [bug] Possible to upload an invalid image file
**Describe the bug**
For research updates (and also certainly for every other instance of file upload), it's possible to upload an invalid image. Neither the field or form highlight that there's a problem and allow the form to be submitted.

By invalid I mean a file type which won't render (e.g. HEIC) or a file with a JPG/PNG extension that (for whatever reason) rendering.

**To Reproduce**
Steps to reproduce the behaviour:

1. Create a research topic.
3. Create a research update.
4. Add a 'wrong' image'.
6. See that it shows an empty space where the image should render.
7. Submit the form with the invalid image
8. See that empty space rendering on the research page.

**Expected behaviour**
The file shouldn't upload in the first place!

**Screenshots**
empty-image
(At time of writting) [update 8](https://community.preciousplastic.com/research/converting-a-small-milling-machine-to-cnc-for-molds-making)

empty-2

**Additional context**
1. Could be related to the Compressor plugin added to Uppy last month.
2. Unit and feature tests should be updated.

### PR DESCRIPTION
**Pull request by iSCJT**
## PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)
- [ ] - Unit and e2e tests for the changes that have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix (fixes an issue)
- [x] Feature (adds functionality)
- [ ] Code style update
- [ ] Refactoring (no functional changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## What is the current behavior?

ImageInput component accepts all files and performs no checks for file integrity.

## What is the new behavior?

1. HEIC files will now be converted to .jpg on upload so that they work across different browsers.
2. Accepted image files limited to '.jpeg', '.jpg', '.png', '.gif', '.svg', '.webp' (safari will auto convert heic files when the accepted mime types are stricter than image/*)
3. Basic check for file integrity using the file reader api - modal displays error message if image fail check, could do with some design work.
4. Images compressed client side before upload, currently set to 60% quality which looks ok to me on my test files but we might want to test some more images that are high quality.  Any png's over 5mb will be converted to jpg and compressed, we can adjust this threshold if needed.
5. Removed the uppy compressor plugin from File Input as it's not used

image

Further work

I ran out of time but will come back to the following bits:

- There's some additional refactoring that can happen with the ImageInput component, specifically the ImageConverter and ImageConverterList components can be simplified / removed.
- Image cropping / editing pre upload.
- Need to add tests, will try and do this tomorrow.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Git Issues

Closes #3755 

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: src/common/Form/ImageInput/ImageInput.tsx [hunk 1]
```diff
@@ -1,54 +1,69 @@
...
+  const onDrop = async (selectedImage) => {
+    try {
+      await imageValid(selectedImage[0])
+      setImageCorrupt(false)
+
+      try {
+        const compressedImage = await compressImage(selectedImage[0])
+        selectedImage[0] = compressedImage
+      } catch (compressionError) {
+        window.console.error(
```
Comment (mariojsnunes) [inline src/common/Form/ImageInput/ImageInput.tsx:48]: better just use the console.error, and add the eslint comment above to ignore the error?

File: src/common/Form/ImageInput/ImageInput.tsx [hunk 2]
```diff
@@ -1,54 +1,69 @@
...
+  const { dataTestId, onFilesChange, value } = props
   const [inputFiles, setInputFiles] = useState<File[]>([])
   const [convertedFiles, setConvertedFiles] = useState<IConvertedFileMeta[]>([])
   const [presentFiles, setPresentFiles] = useState<IMultipleInputValue>(
     getPresentFiles(value),
   )
-
-  const onDrop = (inputFiles) => {
-    setInputFiles(inputFiles)
+  const [imageCorrupt, setImageCorrupt] = useState<boolean>(false)
```
Comment (mariojsnunes) [inline src/common/Form/ImageInput/ImageInput.tsx:36]: I'd rather have it named isImageCorrupt
the boolean type is trivially inferred, just like the line below, no need for it

**Review by mariojsnunes — COMMENTED** [review]
looking good! added some minor comments.

**Comment by iSCJT** [comment]
@benfurber @mariojsnunes sorry guys, I didn't get the tests done in time and I'm off on a trek tomorrow so will be without my laptop till probably 2nd week of August. I'm going to take the draft tag off of this as what's in it should work fine.

I'll pick the tests up when I'm back, feel free to assign an issue to me or something.

**Comment by benfurber** [comment]
Awesome! Looking good. I think it's OK to wait until you have the time to right the tests. Especially working out how to do a feature test for it. 

Do you know about the `ImageCrop` component? I wanted to implement it but struggled in time to wait out when in the flow to add it without having to refactor everything.

**Comment by benfurber** [comment]
Picking this up to get it merged in.

**Comment by benfurber** [comment]
Done a quick test on the preview branch that if you try to upload an image with the wrong extension, the message appears.

**Pull request merged**